compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -266,6 +266,22 @@ object MainGenericRunner {
           run(settings.withExecuteMode(ExecuteMode.Run))
         else
           run(settings.withExecuteMode(ExecuteMode.Repl))
+    end run
+
+    val ranByCoursierBootstrap =
+      sys.props.isDefinedAt("coursier.mainJar")
+      || sys.props.get("bootstrap.mainClass").contains("dotty.tools.MainGenericRunner")
+
+    val silenced = sys.props.get("scala.use_legacy_launcher") == Some("true")
```

Comment:
We may want to document this prop clearly somewhere later, once we release this.
I'm also thinking an env variable could potentially be useful?
`LEGACY_SCALA_RUNNER=true` or smth.